CMakeLists.txt

```diff
@@ -9,38 +9,46 @@ project(
     LANGUAGES CXX
 )
 
-# [CMAKE.SKIP_TESTS]
+enable_testing()
+
 option(
     BEMAN_EXEMPLAR_BUILD_TESTS
-    "Enable building tests and test infrastructure. Default: ON. Values: { ON, OFF }."
+    "Enable building tests and test infrastructure."
     ${PROJECT_IS_TOP_LEVEL}
 )
 
-# [CMAKE.SKIP_EXAMPLES]
 option(
     BEMAN_EXEMPLAR_BUILD_EXAMPLES
-    "Enable building examples. Default: ON. Values: { ON, OFF }."
+    "Enable building examples."
+    ${PROJECT_IS_TOP_LEVEL}
+)
+
+option(
+    BEMAN_EXEMPLAR_FETCH_GOOGLETEST
+    "Enable downloading and building googletest from source."
     ${PROJECT_IS_TOP_LEVEL}
 )
 
 include(FetchContent)
 include(GNUInstallDirs)
 
 if(BEMAN_EXEMPLAR_BUILD_TESTS)
-    enable_testing()
-
-    # Fetch GoogleTest
-    FetchContent_Declare(
-        googletest
-        GIT_REPOSITORY https://github.com/google/googletest.git
-        GIT_TAG
-            f8d7d77c06936315286eb55f8de22cd23c188571 # release-1.14.0
-        EXCLUDE_FROM_ALL
-    )
-    block()
-        set(INSTALL_GTEST OFF) # Disable GoogleTest installation
-        FetchContent_MakeAvailable(googletest)
-    endblock()
+    if(BEMAN_EXEMPLAR_FETCH_GOOGLETEST)
+        # Fetch GoogleTest
+        FetchContent_Declare(
+            googletest
+            GIT_REPOSITORY https://github.com/google/googletest.git
+            GIT_TAG
+                f8d7d77c06936315286eb55f8de22cd23c188571 # release-1.14.0
+            EXCLUDE_FROM_ALL
+        )
+        block()
+            set(INSTALL_GTEST OFF) # Disable GoogleTest installation
+            FetchContent_MakeAvailable(googletest)
+        endblock()
+    else()
+        find_package(GTest REQUIRED)
```

Comment:
Is there never a case where we don't need google test?  I'm actually hoping projects use something else mostly

Reply:
Projects that test another way wouldn't need googletest, I suppose. In particular, projects providing entirely constexpr features wouldn't benefit from a framework like googletest, so the extra dependency would get removed. Also, some libraries would be too low level to justify a dependency on a test framework that pulls in vectors and serializers and such.

I would still expect a CMake test target to exercise a "test suite" of some sort, even if that's a series of example main functions that get compiled and executed.

Reply:
> entirely constexpr features wouldn't benefit from a framework 

Well, in my experience with this in boost date-time I still had runtime tests along with the constexpr validation tests -- of course in that case the runtime tests came first and constexpr later.  Still, with a trust-but-verify thinking I'd still want both compile and runtime covered -- but I wouldn't need/use the massive google test for simple test cases.

At the moment I'm not sure how to resolve the issue -- perhaps we leave a comment that says 'replace test framework here' -- not sure.

Reply:
I think you have valid concerns, Jeff. Let's keep iterating until we have something we're happy with.

That being said, the current PR is entirely about not getting the build to hang given the dependency. Are OK branching off another discussion so we can resolve the bug noted in the meantime?